aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -1139,7 +1139,15 @@ Tensor & index_select_out_cpu_(const Tensor & self, int64_t dim, const Tensor &
   auto index_contig = index.contiguous();
 
   if (self.dim() > 1) {
-    if (numel == 0 || self.numel() == 0) {
+    if (numel == 0) {
+      return result;
+    }
+    if (self.numel() == 0) {
+      AT_DISPATCH_INDEX_TYPES(
+      index_contig.scalar_type(), "index_select_empty_self_bound_check", [&]() {
+        const auto* idxs = index_contig.data_ptr<index_t>();
+        check_indexarray_range<index_t>(idxs, numel, self.size(dim));
+      });
```

Comment:
This feels like it can be more optimized. If the selection dimension is size 0 and there is a nonzero number of indices, the indices are guaranteed to be out of bounds and no checks are necessary, we should just immediately error

Reply:
Thank you for reviewing! Yes, will add the optimization when selection dim is 0. The cuda error seems to be from the new test case, will fix it.